tests/executables_src/testScalarRegisterAccessor.cpp

```diff
@@ -288,3 +324,48 @@ BOOST_AUTO_TEST_CASE(testUniqueID) {
 
   device.close();
 }
+
+BOOST_AUTO_TEST_CASE(testWriteIfDifferent) {
+  setDMapFilePath("dummies.dmap");
+  std::cout << "testRegisterAccessor" << std::endl;
+
+  Device device;
+  device.open("(WriteCountingDummy?map=goodMapFile.map)");
+  auto backend = boost::dynamic_pointer_cast<WriteCountingBackend>(
+      BackendFactory::getInstance().createBackend("(WriteCountingDummy?map=goodMapFile.map)"));
+  BOOST_CHECK(backend != NULL);
+
+  // obtain register accessor with integral type
+  ScalarRegisterAccessor<int> accessor = device.getScalarRegisterAccessor<int>("APP0/WORD_STATUS");
+  BOOST_CHECK(accessor.isReadOnly() == false);
+  BOOST_CHECK(accessor.isReadable());
+  BOOST_CHECK(accessor.isWriteable());
+
+  // dummy register accessor for comparison
+  DummyRegisterAccessor<int> dummy(backend.get(), "APP0", "WORD_STATUS");
+
+  // Inital write and writeIfDifferent with same Value
+  accessor = 501;
+  accessor.write();
+  size_t counterBefore = backend->writeCount;
+  accessor.writeIfDifferent(501); // should not write
+  size_t counterAfter = backend->writeCount;
+  BOOST_CHECK(counterBefore == counterAfter);
+
+  // writeIfDifferent with different Value
+  counterBefore = backend->writeCount;
+  accessor.writeIfDifferent(502); // should write
+  counterAfter = backend->writeCount;
+  BOOST_CHECK(counterBefore != counterAfter);
```

Comment:
Probably ok. Maybe a more precise check (counterAfter > counterBefore) or even (counterAfter = counterBefore +1) would be better.